modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-okhttp/infrastructure/ApiClient.kt.mustache

```diff
@@ -326,7 +333,16 @@ import com.squareup.moshi.adapter
             headers.forEach { header -> addHeader(header.key, header.value) }
         }.build()
 
-        val response = client.newCall(request).execute()
+        val response: Response = suspendCoroutine { continuation ->
+            client.newCall(request).enqueue(object : Callback {
+                override fun onFailure(call: Call, e: IOException) {
+                    continuation.resumeWithException(e)
```

Comment:
If the call is cancelled because the coroutine is cancelled as per your next comment, the resumeWithException shouldn't do anything.
On the other hand, if the call is ever cancelled for another reason, you do want to report the exception and resume, or else the coroutine will never terminate.
See https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/suspend-cancellable-coroutine.html

Reply:
Thanks for the explanation, looks good to me 👍